packages/validator/contracts/SingleClaimantExecutor.sol

```diff
@@ -0,0 +1,40 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "./BaseExecutor.sol";
+import "@openzeppelin/contracts/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/contracts/utils/Base64.sol";
+
+/**
+ * @dev An abstract implementation of an Executor that only allows claims with unique clamaints. 
+ *      To use, subclass and override `executeClaim` and `metadata`, being sure to call `super` inside `executeClaim` before doing anything else.
+ */
+abstract contract SingleClaimantExecutor is BaseExecutor {
+
+    // claimant -> beneficiary
+    mapping (address => address) public claimants;
+    
+    error AlreadyClaimed();
+
+    constructor(address _validator) BaseExecutor(_validator) { }
+
+    /**
+     * @dev Executes a claim that has been verified by the `ValidatorRegistry`. Implementers must check that this function
+     *      was called by a registry they recognise, and that any conditions in claimData such as replay protection are met
+     *      before acting on the request.
+     * @param issuer The account that issued the claim.
+     * @param claimant The account that is entitled to make the claim.
+     * @param beneficiary The account that the claim should benefit.
+     * @param claimData Claim data provided by the issuer.
+     * @param executorData Contextual information stored on the ValidatorRegistry for this issuer.
+     */
+    function executeClaim(address issuer, address claimant, address beneficiary, bytes calldata claimData, bytes calldata executorData) public virtual override {
+        super.executeClaim(issuer, claimant, beneficiary, claimData, executorData);
+
+        // if claimant was already used
+        if(claimants[claimant] != address(0)) {
+            revert AlreadyClaimed();
+        }
+        claimants[claimant] = beneficiary;
+    }    
```

Comment:
Sorry, I gave bad advice earlier on deleting the metadata function stub; can you instead implement it in the same fashion as it is in HighestNonceExecutor? https://github.com/Arachnid/shibboleth/blob/main/packages/validator/contracts/HighestNonceExecutor.sol

Reply:
To do it in the similar fashion, we need to pass `claimant` to metadata function. There are 2 options to do it: 
A. Add `claimant` to `claimdata` (similar to how nonce is added in the HighestNonceExecutor
B. Add `claimant` param to `metadata` function. So, it can be implemented like this: 

```
function metadata(address /*issuer*/, address claimant, bytes calldata /*claimData*/, bytes calldata /*executorData*/) public override virtual view returns(string memory) {
  if(claimants[claimant] != address(0)) {
    return string(abi.encodePacked(
                                    "data:application/json;base64,",
                                     Base64.encode("{\"valid\":false,\"error\":\"Code already claimed..\"}")
                                     ));
  }
  return "";
}
```
Considering we're thinking to make `claimant` to be the ID, it seems that option B makes sense. What do you think?

Reply:
I think option B, mirroring the changes you made to the other function, makes the most sense.

Reply:
@Arachnid done.